drivers/FlashIAP.cpp

```diff
@@ -117,10 +144,19 @@ bool FlashIAP::is_aligned_to_sector(uint32_t addr, uint32_t size)
 
 int FlashIAP::erase(uint32_t addr, uint32_t size)
 {
-    uint32_t current_sector_size = 0UL;
+    uint32_t current_sector_size;
+    uint32_t flash_size = flash_get_size(&_flash);
+    uint32_t flash_start_addr = flash_get_start_address(&_flash);
+    uint32_t flash_end_addr = flash_start_addr + flash_size;
+    uint32_t erase_end_addr = addr + size;
 
-    if (!is_aligned_to_sector(addr, size)) {
+    if (erase_end_addr > flash_end_addr) {
         return -1;
+    } else if (erase_end_addr < flash_end_addr){
+        uint32_t following_sector_size = flash_get_sector_size(&_flash, erase_end_addr);
```

Comment:
is this "following" sector or the "last sector size to be erased"? From the code I read it as it is the last sector to be erased.

Reply:
@LiyouZhou please review

Reply:
No. Following means the one after this range. Couldn't find a better name (any recommendation would be welcome).

Reply:
We are making the assumption that: flash sector always start on an address that is aligned to the flash sector size. This is true for all platfroms that I know but I don't know if it is generally true.

I think this will work for the issue i raised. Can we have a testcase to make sure there is no regression?

Reply:
I also never saw cases where sector addresses weren't aligned to their own sizes. This is pretty easy to fix (subtract flash start from relevant cases). Not sure it justifies a special test (if I understood you correctly).

Reply:
I meant a test for trying to erase 2 sectors of different sizes at once. basically the situation i describe here: https://github.com/ARMmbed/mbed-os/issues/6077

Reply:
Oh I see. The NUCLEO_F410RB has two last sectors having different sizes (16KB  and 64KB). I can change my new test to erase the last two sectors in one call, and this would cover the case.

Reply:
Fixed the test to erase the two last sectors in one call. This will now check your described scenarios in boards whose last two sector sizes are different (there are a few).

Reply:
@LiyouZhou Can you review the latest changes and leave feedback here?

Reply:
@LiyouZhou Would appreciate a quick review, as it seems your review is all that's needed to apply this PR.